proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -252,6 +252,11 @@ public void connect(ServerInfo info, final Callback<Boolean> callback, final boo
             {
                 callback.done( false, null );
             }
+            if ( server == null )
+            {
+                String s = bungee.getTranslation( "connect_kick", info.getName() );
+                disconnect( s.substring( 0, s.length() - 1 ) ); // Remove the colon at end of string
```

Comment:
When I wrote this, for some reason I decided against adding a reason parameter for some reason so the message would have shown as 'Kicked whilst connecting to x:' which would have looked strange with the colon at the end.

I agree that your proposed method below is a better solution however.